vault/resource_database_secret_backend_connection_test.go

```diff
@@ -739,6 +740,7 @@ func TestAccDatabaseSecretBackendConnection_postgresql(t *testing.T) {
 					resource.TestCheckResourceAttr(testDefaultDatabaseSecretBackendResource, "postgresql.0.max_idle_connections", "0"),
 					resource.TestCheckResourceAttr(testDefaultDatabaseSecretBackendResource, "postgresql.0.max_connection_lifetime", "0"),
 					resource.TestCheckResourceAttr(testDefaultDatabaseSecretBackendResource, "postgresql.0.username", username),
+					resource.TestCheckResourceAttr(testDefaultDatabaseSecretBackendResource, "postgresql.0.password", password),
 					resource.TestCheckResourceAttr(testDefaultDatabaseSecretBackendResource, "postgresql.0.disable_escaping", "true"),
 					resource.TestCheckResourceAttr(testDefaultDatabaseSecretBackendResource, "postgresql.0.username_template", userTempl),
 				),
```

Comment:
I wasn't sure how to test the exact problematic scenario automatically and if it is worth an explicit test or is too niche. I did not see tests that validate the plan has no changes to apply following other operations.

Reply:
Hmm, I think test parameters such as `PlanOnly` or `ExpectNonEmptyPlan` might help you achieve this test structure and confirm that a plan was successfully validated/invalidated either with/without a drift. Here is an example of this being used in the codebase — [example](https://github.com/hashicorp/terraform-provider-vault/blob/main/vault/resource_token_test.go#L203-L212)

Reply:
Got it to work, was looking for a expectEmptyPlan option but the framework uses the double negative of `ExpectNonEmptyPlan = false` to achieve this.